backend/src/api/v1/mentorShip/services/userServices.ts

```diff
@@ -2,39 +2,31 @@ import { ObjectId } from 'mongodb'
 import { getCollection } from '../../../../database/db'
 import { genSaltSync, hashSync } from 'bcrypt-ts'
 import { uuid } from 'uuidv4'
-
-interface userDetail {
-    firstName: string
-    lastName: string
-    email: string
-    username: string
-    password: string
-    location: string
-}
+import { userDetails, userInfo } from './types'
 
 export const getAllDataService = async () => {
-    const collection = await getCollection('users')
-    const allData = await collection?.find({}).toArray()
+    const userList = await getCollection('users')
+    const allData = await userList?.find({}).toArray()
 
     console.log('collection', allData)
 
     return allData
 }
 
-export const getSingleUser = async (id: string) => {
-    const collection = await getCollection('users')
-    const userData = await collection?.findOne({
-        _id: new ObjectId(id)
+export const getSingleUser = async (args: userInfo) => {
+    const userList = await getCollection('users')
+    const userData = await userList?.findOne({
+        id: new ObjectId(args.id)
     })
 
```

Comment:
We need to fetch a user by their email as it is the only unique identifier we have